File: backend/src/models/AIProvider.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IAIProvider extends Document {
    name: string;
    baseURL: string;
    apiKey: string;
    modelName: string;
    embeddingApiKey?: string;  // Embedding 专用 API Key，留空则复用 apiKey
    embeddingModel?: string;    // Embedding 模型名（如 embo-01，默认用 text-embedding-ada-002）
    embeddingBaseURL?: string;  // Embedding 专用接口（如 Minimax 用不同地址）
    embeddingDimensions?: number; // Embedding 维度（如 DashScope v4 可显式指定 1536）
    groupId?: string;           // 部分 provider 需要额外参数（如 Minimax）
    enabled: boolean;
    isDefault: boolean;
    createdAt: Date;
    updatedAt: Date;
}

const aiProviderSchema = new Schema<IAIProvider>(
    {
        name: {
            type: String,
            required: true,
            unique: true,
        },
        baseURL: {
            type: String,
            required: true,
        },
        apiKey: {
            type: String,
            required: true,
        },
        embeddingApiKey: {
            type: String,
            required: false,
        },
        modelName: {
            type: String,
            required: true,
        },
        embeddingBaseURL: {
            type: String,
            required: false,
        },
        embeddingModel: {
            type: String,
            required: false,
        },
        embeddingDimensions: {
            type: Number,
            required: false,
            min: 1,
        },
        groupId: {
            type: String,
            required: false,
        },
        enabled: {
            type: Boolean,
            default: true,
        },
        isDefault: {
            type: Boolean,
            default: false,
        },
    },
    {
        timestamps: true,
    }
);

export default mongoose.model<IAIProvider>('AIProvider', aiProviderSchema);

File: mobile/lib/data/models/ai_provider.dart
class AIProvider {
  final String id;
  final String name;
  final String baseURL;
  final String modelName;
  final String? embeddingApiKey;
  final String? embeddingModel;
  final String? embeddingBaseURL;
  final int? embeddingDimensions;
  final String? groupId;
  final bool enabled;
  final bool isDefault;
  final String createdAt;

  AIProvider({
    required this.id,
    required this.name,
    required this.baseURL,
    required this.modelName,
    this.embeddingApiKey,
    this.embeddingModel,
    this.embeddingBaseURL,
    this.embeddingDimensions,
    this.groupId,
    required this.enabled,
    required this.isDefault,
    required this.createdAt,
  });

  factory AIProvider.fromJson(Map<String, dynamic> json) {
    return AIProvider(
      id: json['_id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      baseURL: json['baseURL'] as String? ?? '',
      modelName: json['modelName'] as String? ?? '',
      embeddingApiKey: json['embeddingApiKey'] as String?,
      embeddingModel: json['embeddingModel'] as String?,
      embeddingBaseURL: json['embeddingBaseURL'] as String?,
      embeddingDimensions: json['embeddingDimensions'] as int?,
      groupId: json['groupId'] as String?,
      enabled: json['enabled'] as bool? ?? true,
      isDefault: json['isDefault'] as bool? ?? false,
      createdAt: json['createdAt'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'baseURL': baseURL,
      'modelName': modelName,
      if (embeddingApiKey != null) 'embeddingApiKey': embeddingApiKey,
      if (embeddingModel != null) 'embeddingModel': embeddingModel,
      if (embeddingBaseURL != null) 'embeddingBaseURL': embeddingBaseURL,
      if (embeddingDimensions != null) 'embeddingDimensions': embeddingDimensions,
      if (groupId != null) 'groupId': groupId,
      'enabled': enabled,
      'isDefault': isDefault,
    };
  }
}
